src/graphics/device.js

```diff
@@ -1785,6 +1785,12 @@ Object.assign(GraphicsDevice.prototype, {
     },
 
     uploadTexture: function (texture) {
+        // #ifdef DEBUG
+        if (!texture.device) {
+            console.error("attempting to use a texture that has been destroyed.");
```

Comment:
Isn't this going to potentially spam the JS console since this function is called a lot?

Reply:
yeah possibly...